### PR TITLE
Added project feature

### PR DESCRIPTION
**Pull request by malikalhack**

### CODE REVIEW

File: UsbVirtualPort.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
+﻿#include <Windows.h>
+#include <stdio.h>
+#include <cstdlib>
+
+HANDLE hCommR, hCommW;   // Handle to the Serial ports
+
+int finish_program(int status) {
+    CloseHandle(hCommR); //Closing the serial port for reading
+    CloseHandle(hCommW); //Closing the serial port for reading
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: Fix the typo

File: UsbVirtualPort.cpp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    getchar();
+    return status;
+}
+
+int main() {
+    char   rComPortName[] = "\\\\.\\COM3";  // Serial port name to read
+    char   wComPortName[] = "\\\\.\\COM4";  // Serial port name to write
+    DWORD  dwEventMask;                     // Event mask to trigger
+    char   TempChar;                        // Temperory Character
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: Why is it here?

File: UsbVirtualPort.cpp [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    char   SerialBuffer[256];               // Buffer Containing Rxed Data
+    DWORD  NoBytesRead;                     // Bytes read by ReadFile()
+
+    char   lpBuffer[] = "Hello world!";     // lpBuffer should be char or byte array
+    DWORD  dNoOFBytestoWrite;               // Number of bytes to write to the port
+    DWORD  dNoOfBytesWritten = 0;           // Number of bytes written to the port
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: What is "No"?

File: UsbVirtualPort.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    ) {
+        printf("Error >> The SetCommTimeouts() function ended with an error\n");
+        return finish_program(EXIT_FAILURE);
+    }
+#ifdef _DEBUG
+    else printf("Serial Ports Timeout Setting Succeeded\n");
+#endif // _DEBUG
+
+/*--------------------------- Setting Receive Mask ---------------------------*/
+    //Configure Windows to Monitor the serial device for Character Reception
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: The phrase is not clear. Rephrase.

File: UsbVirtualPort.cpp [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    else printf("Communication mask port setup successful\n");
+#endif // _DEBUG
+
+/*-------------------- Writing a Character to Serial Port --------------------*/
+    if (
+        WriteFile(
+            hCommW,             // Handle to the Serialport
+            lpBuffer,           // Data to be written to the port 
+            dNoOFBytestoWrite,  // No of bytes to write into the port
+            &dNoOfBytesWritten, // No of bytes written to the port
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: The "No" again

File: UsbVirtualPort.cpp [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+/*------------------------- Setting WaitComm() Event -------------------------*/
+/*----------- Program will Wait here till a Character is received ------------*/
+    printf("\n============== Waiting for Data Reception ===============\n\n");
+    //Wait for the character to be received
+    if (WaitCommEvent(hCommR, &dwEventMask, NULL) == FALSE) {
+        printf("Error >> The WaitCommEvent() function ended with an error\n");
+        return finish_program(EXIT_FAILURE);
+    }
+    else {
+        //Read the RXed data using ReadFile();
```
Comment (malikalhack2) [inline UsbVirtualPort.cpp]: Write the word in full. Don't be misled.

File: UsbVirtualPort.cpp (hunk 1 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

File: UsbVirtualPort.cpp (hunk 2 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Will replace

File: UsbVirtualPort.cpp (hunk 3 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: "No" equals "Number". Will fix.

File: UsbVirtualPort.cpp (hunk 4 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Will fix

File: UsbVirtualPort.cpp (hunk 5 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Will fix

File: UsbVirtualPort.cpp (hunk 6 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Will fix

**Review by malikalhack2 — APPROVED** [review]

File: UsbVirtualPort.cpp (hunk 2 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

File: UsbVirtualPort.cpp (hunk 3 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

File: UsbVirtualPort.cpp (hunk 4 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

File: UsbVirtualPort.cpp (hunk 5 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

File: UsbVirtualPort.cpp (hunk 6 above)
Comment (malikalhack) [reply UsbVirtualPort.cpp]: Fixed

**Pull request merged**